linux/mount.go

```diff
@@ -914,10 +928,13 @@ func SetupFilesystem(device *model.Device, filesystemType string) error {
 		return nil
 	}
 
-	log.Tracef("Creating filesystem %s on device path %s", filesystemType, device.AltFullPathName)
-	if err := RetryCreateFileSystem(device.AltFullPathName, filesystemType); err != nil {
-		log.Errorf("Failed to create filesystem %s on device with path %s", filesystemType, device.AltFullPathName)
-		return err
+	// For clone case, filesystem will be present
+	if !device.FilesystemPresent {
```

Comment:
This flag is set only when EncryptionKey is set? What about the existing scenario where `volume.EncryptionKey == ""` This needs to handle both the cases.

Reply:
This flag is not required anymore. It was added to disallow mkfs on the cloned-volume. However, after the lines 916-919 were added, this flag has become redundant:

https://github.com/hpe-storage/common-host-libs/blob/6ab488305b0c4725ea9a873a4ceaf2066ee7cfab/linux/mount.go#L916